### DISCUSSION TITLE
Why I can't select text in disassebly view?

### DISCUSSION THREAD

**Discussion by FeelUsM** [microsoft/vscode-cpptools#12077] [General]
Why is the disassembly view window made not only “read-only” but completely static so that I can’t select the text anymore?
I’m interested in the technical details, why it was impossible to implement this window as “read-only” (so that it would be possible to select and copy text from there)

**Comment by bobbrow** [comment]
@xisui-MSFT do you know the answer?

@FeelUsM we helped implement this years ago, but I don't remember the specifics here. The code for this view lives in VS Code itself, so if we open a request to unblock selections, it would need to be in their repo.

**Comment by xisui-MSFT** [comment]
It's implemented as an infinitely scroll-able list, which loads the content when scrolling. It can't implemented as a text file because it could be very large for large programs.

Because of the design limitations of lists in VS Code, when a roll of a list is not on the screen, it will be disposed, and therefore can't be selected (or, will be unselected when you scroll away). So you won't able to copy more than what's shown on your screen. This is the same as the output window.

That said, I think we implemented the texts in it as selectable. There might be some changes after. You can open a ticket in VS Code repo, but you'll unlikely get the perfect copy-paste experience you would like to see.
